### PR TITLE
Clean up route tables and NAT for quay

### PR DESCRIPTION
**Pull request by meatballhat**
Targeting the now-very-much-overgrown #28

### CODE REVIEW

**Comment by meatballhat** [comment]
@henrikhodne r?

**Comment by meatballhat** [comment]
@igorwwwwwwwwwwwwwwwwwwww r?

File: aws-shared-1/main.tf [hunk 1]
```diff
@@ -8,30 +13,66 @@ variable "workers_org_subnet_1e" { default = "10.10.6.0/24" }
...
+    protocol = "-1"
+    cidr_blocks = ["0.0.0.0/0"]
+  }
+}
+
+resource "aws_instance" "nat_quay" {
+  ami = "${var.nat_ami}"
+  instance_type = "c3.2xlarge"
+  source_dest_check = false
+  subnet_id = "${module.aws_az_1b.public_subnet_id}"
```
Comment (meatballhat) [inline aws-shared-1/main.tf]: I figured pinning to az 1b was acceptable :shrug:

File: bin/generate-aws-route-table-public [hunk 2]
```diff
@@ -0,0 +1,33 @@
+#!/usr/bin/env bash
+
+set -o errexit
+
+main() {
+  cat <<EOF
+# WARNING: this file is generated!
```
Comment (henrikhodne) [inline bin/generate-aws-route-table-public]: Would it be useful to also specify that it's this script that generated the file?

**Comment by meatballhat** [comment]
I'm not 100% sure this will work, either, and given that I'm creating a new main route table association, I think this means I need to backfill some more of the default routes such as `10.0.0.0/8` local.

If dropping the quay NAT is an option, I'm all for it.

**Comment by meatballhat** [comment]
Oh look! https://www.terraform.io/docs/providers/aws/r/route_table.html

> Note that the default route, mapping the VPC's CIDR block to "local", is created implicitly and cannot be specified.

**Pull request merged**